### ISSUE TITLE
TS6133 error on deconstruction variable in for await loop 

### ISSUE THREAD

**Issue by viyaha** [microsoft/TypeScript#29796]
**TypeScript Version:** 3.4.0-dev.20190207

**Search Terms:**
for await
async 
AsyncIterableIterator
AsyncIterable
TS6133

**Code**

```ts
async function* customGenerator(): AsyncIterableIterator<any>{
    yield "!";
}


async function bar(){
    const asyncIterableIterator = customGenerator();
    for await (const {} of asyncIterableIterator){
        console.log("!");
    }
}

bar().catch(() => {})
```

**Expected behavior:**
Typescript compiles without TS6133 (unused variable) error.

**Actual behavior:**
![image](https://user-images.githubusercontent.com/44086665/52414028-74fa1500-2ae3-11e9-8147-380f70a6f242.png)

**Comment by rbuckton** [comment]
@viyaha: Can you verify whether this issue still exists against `typescript@rc`? I am unable to reproduce the issue in TypeScript 3.5, so it may have been addressed by some other change.

**Comment by viyaha** [comment]
@rbuckton: The invocation of "tsc --project ." does not throw any error anymore (3.5.0-dev.20190417).
==> The issue seems to be fixed.
